doc/SONiC_local_users_password_reset_hld.md

```diff
@@ -0,0 +1,285 @@
+|   short reset button press  |   Pressing the physical reset button on the switch less than 15 seconds              |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+### 
+
+# Requirements Overview
+
+The feature implementation shall support the following:
+
+    1. The OS will restore the local users' configurations, by deleting non-default users and restoring default passwords for default users and expiring these passwords due to security concerns during **init**.
+    2. The default behavior to trigger this functionality is done by long reset button press, but each vendor can add his desired implementation to when to trigger this feature.
+    3. Feature should be enabled or disabled by setting compilation flag to true or false.
+    4. Feature can be disabled and enabled using CLI command.
+    5. The feature is disabled by default.
+    6. The service introduced as part of this feature will run if the platform has capability of long reset support.
+    7. The feature will not affect warm and fast reboot 
+
+
+### 
+
+# Architecture Design
+
+In this feature, we want to add new systemd service ```reset-local-users-passwords.service``` to the init flow. This new service is expected to run on the system during boot and before any connection is open to the switch which includes SSH and serial connection as well and perform restoring local users' configurations, by deleting non-default users and restoring default passwords for default users and expiring these passwords due to security conecerns.
+
+![](./reset-local-users-passwords-init.png)
+
+The service is dependent on the ```database.service``` for reading the configured feature state, so it needs to run after it during boot. As well as it needs to run before any connection is open and this includes all the following services: `sshd.service`, `getty.target`, `systemd-logind.service` and `serial-getty@ttyS0.service`.
+
+### 
+
+# High-Level Design
+
+This feature will be a built-in SONiC feature. There will be three main files to consider in this feature:
+
+1. ```src/sonic-platform-common/sonic_platform_base/reset_local_users_passwords_base.py``` - The default behavior implementation will reside in this file, including when to trigger the feature and how to reset the local users' configurations.
+2. ```platform/<vendor-path>/sonic_platform/reset_local_users_passwords.py``` - The vendor specifc implementation of the feature, each vendor can decide what is the trigger cause to start the functionality and how it is implemented.
+3. ```src/sonic-host-services/scripts/reset-local-users-passwords``` - The python file that will be called on service start during init that imports the vendor's specific implementation.
+
+The default behavior will delete non-default users and restore the original passwords of the default users and expire them based on the content of the file of ```/etc/sonic/default_users.json``` on long reboot press.
```

Comment:
@azmy98 , can you give more detail about how to restore default password?

According to this HLD: https://github.com/sonic-net/SONiC/blob/master/doc/reset_factory/ResetFactoryHLD.md?plain=1

The "/etc/sonic/default_users.json" only contains **hashed** passwords. so, it's not possible restore default user password from this file.

Reply:
@liuh-80, It is possible sir. You can use the `chpasswd -e` shell command for this purpose. Please refer to an example in the `files/image_config/reset-factory/reset-factory` file under `reset_users()` function.

Reply:
#closed